iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/compression/WALCompressionTest.java

```diff
@@ -18,37 +18,392 @@
+        Assert.assertArrayEquals(nodeBuffer1.array(), nodeBuffer2.array());
+      }
+    }
+  }
+
+  @Test
+  public void testUncompressedWALStructure()
+      throws QueryProcessException, IllegalPathException, IOException {
+    PublicBAOS baos = new PublicBAOS();
+    DataOutputStream dataOutputStream = new DataOutputStream(baos);
+    List<InsertRowNode> insertRowNodes = new ArrayList<>();
+    for (int i = 0; i < 100; ++i) {
+      InsertRowNode node = WALTestUtils.getInsertRowNode(devicePath, i);
+      insertRowNodes.add(node);
+      node.serialize(dataOutputStream);
+    }
+    dataOutputStream.close();
+    ByteBuffer buf = ByteBuffer.wrap(baos.toByteArray());
+    // Do not compress it
+    IoTDBDescriptor.getInstance()
+        .getConfig()
+        .setWALCompressionAlgorithm(CompressionType.UNCOMPRESSED);
+    try (WALWriter writer = new WALWriter(walFile)) {
+      buf.position(buf.limit());
+      writer.write(buf);
+    }
+
+    try (DataInputStream dataInputStream =
+        new DataInputStream(new BufferedInputStream(Files.newInputStream(walFile.toPath())))) {
+      byte[] magicStringBytes = new byte[WALWriter.MAGIC_STRING_BYTES];
+      // head magic string
+      dataInputStream.readFully(magicStringBytes);
+      Assert.assertEquals(WALWriter.MAGIC_STRING, new String(magicStringBytes));
+      Assert.assertEquals(
+          CompressionType.UNCOMPRESSED, CompressionType.deserialize(dataInputStream.readByte()));
+      Assert.assertEquals(buf.array().length, dataInputStream.readInt());
+      ByteBuffer dataBuf = ByteBuffer.allocate(buf.array().length);
+      dataInputStream.readFully(dataBuf.array());
+      Assert.assertArrayEquals(buf.array(), dataBuf.array());
+      Assert.assertEquals(
+          new WALSignalEntry(WALEntryType.WAL_FILE_INFO_END_MARKER),
+          WALEntry.deserialize(dataInputStream));
+      ByteBuffer metadataBuf = ByteBuffer.allocate(12 + Integer.BYTES);
+      dataInputStream.readFully(metadataBuf.array());
+      // Tail magic string
+      dataInputStream.readFully(magicStringBytes);
+      Assert.assertEquals(WALWriter.MAGIC_STRING, new String(magicStringBytes));
+    }
+  }
+
+  @Test
+  public void testCompressedWALStructure()
+      throws IOException,
+          QueryProcessException,
+          IllegalPathException,
+          NoSuchFieldException,
+          ClassNotFoundException,
+          IllegalAccessException {
+    PublicBAOS baos = new PublicBAOS();
+    DataOutputStream dataOutputStream = new DataOutputStream(baos);
+    List<InsertRowNode> insertRowNodes = new ArrayList<>();
+    for (int i = 0; i < 100; ++i) {
+      InsertRowNode node = WALTestUtils.getInsertRowNode(devicePath, i);
+      insertRowNodes.add(node);
+      node.serialize(dataOutputStream);
+    }
+    dataOutputStream.close();
+    ByteBuffer buf = ByteBuffer.wrap(baos.toByteArray());
+    // Do not compress it
+    IoTDBDescriptor.getInstance().getConfig().setWALCompressionAlgorithm(CompressionType.LZ4);
```

Comment:
Do not compress it?

Reply:
Fix it. The comment is wrong since it is copied from previous test case.